### PR TITLE
[WIP] Multi-Cloud PUT/GET/DELETE bucket CORS APIs

### PR DESCRIPTION
**Pull request by nguptaopensds**
**What this PR does / why we need it**:
This PR consists of PUT / GET/ DELETE CORS configuration on bucket APIs

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by nguptaopensds** [comment]
Some issue so have to close

**Comment by nguptaopensds** [comment]
This PR consists some changes from PR #176 so raised it as WIP, once #176 is merged then will remove WIP

File: api/pkg/s3/bucketgetCORS.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	// convert back to xml struct
+	getCorsConf := model.Cors{}
+
+	// convert lifecycle rule to xml Rule
+	if bucket.CorsConfiguration != nil {
+		for _, lcRule := range bucket.CorsConfiguration {
+			xmlRule := model.CorsRule{}
+
+			// Convert Allowed methods from json to xml struct
+			xmlRule.AllowedMethods = lcRule.AllowedMethods
```
Comment (himanshuvar) [inline api/pkg/s3/bucketgetCORS.go]: Can you optimize the code by adding a function json to xml for reuse.
xmljsonfunc(x,y) -> xml.x = json.y

File: api/pkg/s3/bucketputCORS.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		err := xml.Unmarshal(body, &bucketcorsConf)
+		if err != nil {
+			response.WriteError(http.StatusInternalServerError, err)
+			return
+		} else {
+			s3CorsPtrArr := make([]*s3.CORSRule, 0)
+			for _, rule := range bucketcorsConf.CorsRules {
+				s3CORSRule := s3.CORSRule{}
+
+				s3CORSRule.Id = rule.Id
```
Comment (himanshuvar) [inline api/pkg/s3/bucketputCORS.go]: Same as here. function can be utilized here.

**Comment by himanshuvar** [comment]
Can you attach the Test Report here

File: api/pkg/s3/bucketgetCORS.go (hunk 1 above)
Comment (nguptaopensds) [reply api/pkg/s3/bucketgetCORS.go]: Sure, I can refine the code by making unmarshal, marshal generic

File: api/pkg/s3/bucketputCORS.go (hunk 2 above)
Comment (nguptaopensds) [reply api/pkg/s3/bucketputCORS.go]: Same as above

**Pull request closed without merging**